__tests__/pino.spec.ts

```diff
@@ -0,0 +1,202 @@
+import { NestFactory } from "@nestjs/core";
+import { Module, Controller, Get, Injectable, Logger as NestjsLogger } from "@nestjs/common";
+import MemoryStream = require("memorystream");
+import * as request from "supertest";
+import { Logger, LoggerModule } from "../src";
+import { platforms } from "./utils/platforms";
+import { fastifyExtraWait } from "./utils/fastifyExtraWait";
+import { parseLogs } from "./utils/logs";
+import { __resetOutOfContextForTests } from "../src/PinoLogger";
+import * as pino from 'pino';
+
+const loggerMethods: [keyof Logger, pino.Level][] = [
+  ["verbose", "trace"],
+  ["debug", "debug"],
+  ["log", "info"],
+  ["warn", "warn"],
+  ["error", "error"]
+];
+
+describe("pino logging", () => {
+  beforeEach(() => __resetOutOfContextForTests());
+
+  for (const PlatformAdapter of platforms) {
+    describe(PlatformAdapter.name, () => {
+      for (const useContext of [true, false]) {
+        describe(`usage via @nestjs/common/Logger ${useContext ? 'with' : 'without'} context`, () => {
+          for (const [loggerMethodName, pinoLevel] of loggerMethods) {
+            describe(loggerMethodName, () => {
+              it("logs strings as `msg` field", async () => {
+                const stream = new MemoryStream();
+                const random = Math.random().toString();
+                let logs = "";
+                stream.on("data", (chunk: string) => {
+                  logs += chunk.toString();
+                });
+
+                @Controller("/")
+                class TestController {
+                  logger = new NestjsLogger(useContext ? TestController.name : undefined);
```

Comment:
I just didn't get it, why are we using original `Logger` from `@nestjs/common` here instead of `Logger` from this library? It's not the case how it was supposed to be used, just looket at your bug example, and there were the same